class.jetpack-gutenberg.php

```diff
@@ -16,63 +78,77 @@ public static function is_gutenberg_available() {
 	}
 
 	/**
-	 * Load Gutenberg assets
+	 * Check whether conditions indicate Gutenberg blocks should be loaded
+	 *
+	 * Loading blocks is enabled by default and may be disabled via filter:
+	 *   add_filter( 'jetpack_gutenberg', '__return_false' );
 	 *
 	 * @since 6.7.0
 	 *
-	 * @return void
+	 * @return bool
 	 */
-	public static function enqueue_block_assets() {
-		if ( ! self::should_load_blocks() ) {
-			return;
+	public static function should_load_blocks() {
+		if ( ! Jetpack::is_active() ) {
+			return false;
 		}
 
-		$rtl = is_rtl() ? '.rtl' : '';
-
 		/**
-		 * Filter to enable serving blocks via CDN
-		 *
-		 * CDN cache is busted once a day or when Jetpack version changes. To customize it:
-		 *   add_filter( 'jetpack_gutenberg_cdn_cache_buster', function( $version ) { return time(); }, 10, 1 );
+		 * Filter to disable Gutenberg blocks
 		 *
 		 * @since 6.5.0
 		 *
-		 * @param bool false Whether to load Gutenberg blocks from CDN
+		 * @param bool true Whether to load Gutenberg blocks
 		 */
-		if ( apply_filters( 'jetpack_gutenberg_cdn', false ) ) {
-			$cdn_base    = 'https://s0.wp.com/wp-content/mu-plugins/jetpack/_inc/blocks';
-			$view_script = "$cdn_base/view.js";
-			$view_style  = "$cdn_base/view$rtl.css";
-
-			/**
-			 * Filter to modify cache busting for Gutenberg block assets loaded from CDN
-			 *
-			 * @since 6.5.0
-			 *
-			 * @param string
-			 */
-			$version = apply_filters( 'jetpack_gutenberg_cdn_cache_buster', sprintf( '%s-%s', gmdate( 'd-m-Y' ), JETPACK__VERSION ) );
-		} else {
-			$view_script = plugins_url( '_inc/blocks/view.js', JETPACK__PLUGIN_FILE );
-			$view_style  = plugins_url( "_inc/blocks/view$rtl.css", JETPACK__PLUGIN_FILE );
-			$version     = Jetpack::is_development_version() && file_exists( JETPACK__PLUGIN_DIR . '_inc/blocks/view.js' )
-				? filemtime( JETPACK__PLUGIN_DIR . '_inc/blocks/view.js' )
-				: JETPACK__VERSION;
+		return (bool) apply_filters( 'jetpack_gutenberg', true );
+	}
+
+	/**
+	 * Only enqueue block assets when needed.
+	 *
+	 * @param string $type slug of the block.
+	 *
+	 * @return void
+	 */
+	public static function load_assets_as_required( $type ) {
+		// Enqueue styles.
+		$style_relative_path = '_inc/blocks/' . $type . '/view' . ( is_rtl() ? '.rtl' : '' ) . '.css';
+		if ( self::block_has_asset( $style_relative_path ) ) {
+			$style_version = self::get_asset_version( $style_relative_path );
+			$view_style    = plugins_url( $style_relative_path, JETPACK__PLUGIN_FILE );
+			wp_enqueue_style( 'jetpack-block-' . $type, $view_style, array(), $style_version );
 		}
 
-		wp_enqueue_script(
-			'jetpack-blocks-view',
-			$view_script,
-			array(
-				'wp-element',
-				'wp-i18n',
-			),
-			$version
-		);
+		// Enqueue script.
+		$script_relative_path = '_inc/blocks/' . $type . '/view.js';
```

Comment:
Since the blocks now live in subdirectories, we will need to update this accordingly:
https://github.com/Automattic/jetpack/blob/20da538a07f111c238f50311e2f27a1605f7c930/gulpfile.babel.js#L255

Reply:
Done in 54940f3cc00f07a1e61ab8b4298fe90ac8cca07e